GiEnJul/ClientApp/src/components/admin/overview/Recipient.tsx

```diff
@@ -3,55 +3,19 @@ import { Search } from "@material-ui/icons";
 import * as React from "react";
 import { useEffect, useState } from "react";
 import Datatable from "../common/RecipientTable";
-import { GiverType } from "./Giver";
+import { GiverType, RecipientType, PersonType } from "./OverviewMacro";
 
-export interface PersonType {
-  partitionKey: string;
-  rowKey: string;
-  wish: string;
-  age: Number;
-  gender: Number;
-}
-export interface RecipientType {
-  contactEmail: string;
-  contactFullName: string;
-  contactPhoneNumber: string;
-  dessert: string;
-  dinner: string;
-  eventName: string;
-  familyMembers: PersonType[];
-  hasConfirmedMatch: Boolean;
-  institution: string;
-  isSuggestedMatch: Boolean;
-  location: string;
-  matchedGiver?: GiverType;
-  note: string;
-  partitionKey: string;
-  referenceId: string;
-  rowKey: string;
-}
+type Props = {
+  data: RecipientType[] | [];
+  handleRecipientChange: (
+    newRecipientRowKey: string,
+    newRecipientPartitionKey: string
+  ) => void;
+};
 
-const Recipient = () => {
-  const [data, setData] = useState<RecipientType[] | []>([]);
+const Recipient: React.FC<Props> = ({ data, handleRecipientChange }) => {
   const [q, setQ] = useState("");
```

Comment:
I think there should be a more descriptive name here. E.g. `query`?

Reply:
agree